File: 0748-largest-number-at-least-twice-of-others/0748-largest-number-at-least-twice-of-others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int maxi=Integer.MIN_VALUE , index=-1;
        for (int i=0;i<nums.length;i++){
            if (nums[i]>maxi){
                index=i;
                maxi=nums[i];
        }}
        for (int i=0;i<nums.length;i++){
            if (nums[i]==maxi) continue;
            if ((nums[i]*2)>maxi) return -1;
        }
        return index;
    }
}

File: 0748-largest-number-at-least-twice-of-others/0748-largest-number-at-least-twice-of-others.py
from typing import List

class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        maxi = float('-inf')
        index = -1
        for i in range(len(nums)):
            if nums[i] > maxi:
                maxi = nums[i]
                index = i
        for i in range(len(nums)):
            if nums[i] == maxi:
                continue
            if nums[i]*2 > maxi:
                return -1
        return index
